legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java

```diff
@@ -597,6 +597,11 @@ public Handlers(PureModel pureModel)
         register("meta::json::schema::mapSchema_String_1__Type_1__DiscriminatorMapping_1_", false, ps -> res("meta::json::schema::DiscriminatorMapping", "one"));
         register("meta::json::schema::discriminateOneOf_Any_1__Any_1__Type_MANY__DiscriminatorMapping_MANY__Boolean_1_", false, ps -> res("Boolean", "one"));
 
+        // getter for execution parameters from execution environment
+        register(m(
+                m(h("meta::legend::service::get_ExecutionEnvironmentInstance_1__String_$0_1$__SingleExecutionParameters_1_", false, ps -> res("meta::legend::service::metamodel::SingleExecutionParameters", "one"), ps -> ps.size() == 2)),
```

Comment:
Why is masterKey optional?

Reply:
changed it to non be optional for both this and single execution parameters